src/seed_tools/utils/study_validation.ts

```diff
@@ -80,11 +88,97 @@ function checkExperiments(study: Study): string[] {
+
+    // Validate features.
+    if (experiment.feature_association !== undefined) {
+      if (
+        experiment.feature_association.enable_feature !== undefined &&
+        experiment.feature_association.enable_feature.length > 0
+      ) {
+        for (const feature of experiment.feature_association.enable_feature) {
+          checkFeatureName(experiment, feature, errors);
+        }
+      }
+      if (
+        experiment.feature_association.disable_feature !== undefined &&
+        experiment.feature_association.disable_feature.length > 0
+      ) {
+        for (const feature of experiment.feature_association.disable_feature) {
+          checkFeatureName(experiment, feature, errors);
+        }
+      }
+      if (experiment.feature_association.forcing_feature_on !== undefined) {
+        checkFeatureName(
+          experiment,
+          experiment.feature_association.forcing_feature_on,
+          errors,
+        );
+      }
+      if (experiment.feature_association.forcing_feature_off !== undefined) {
+        checkFeatureName(
+          experiment,
+          experiment.feature_association.forcing_feature_off,
+          errors,
+        );
+      }
+    }
+
+    // Validate forcing flag.
+    if (experiment.forcing_flag !== undefined) {
+      if (
+        !isStringASCIIWithoutChars(experiment.forcing_flag, '') ||
+        experiment.forcing_flag !== experiment.forcing_flag.toLowerCase()
+      ) {
+        errors.push(
+          `Invalid forcing flag for experiment ${experiment.name}: ${experiment.forcing_flag} (expected lowercase ASCII)`,
+        );
+      }
+    }
+
+    // Validate google_web_experiment_id and google_web_trigger_experiment_id.
+    if (
+      experiment.google_web_experiment_id !== undefined &&
+      experiment.google_web_trigger_experiment_id !== undefined
+    ) {
+      errors.push(
+        `Experiment ${experiment.name} has both google_web_experiment_id and web_trigger_experiment_id`,
+      );
+    }
+
+    // Valiate params.
+    const paramNames = new Set<string>();
+    if (experiment.param.length > 0) {
+      for (const param of experiment.param) {
+        if (param.name === undefined || param.name === '') {
+          errors.push(`Empty param name in experiment ${experiment.name}`);
+          continue;
+        }
+        if (paramNames.has(param.name)) {
+          errors.push(
+            `Duplicate param name: ${param.name} in experiment ${experiment.name}`,
+          );
+        }
+        paramNames.add(param.name);
+      }
+    }
   }
 
+  // Validate default_experiment_name.
+  if (
+    study.default_experiment_name !== undefined &&
+    study.default_experiment_name !== '' &&
+    !experimentNames.has(study.default_experiment_name)
+  ) {
+    errors.push(
+      `Missing default experiment: ${study.default_experiment_name} in ${study.name} study`,
+    );
+  }
+
+  // Validate total probability.
   if (totalProbability !== 100) {
```

Comment:
generally it's not neccessarily true, but probably we can limit ourself to 100 for simplicity
<img width="408" alt="image" src="https://github.com/user-attachments/assets/eebf9a29-66d8-4bdf-8834-796642aa05c4">

Reply:
> We can assume `totalProbability == 100` for Griffin

yes, that's what we did before and I assume we will continue doing. It's just easier to view this as 0-100 percent value.